File: Lab2.1/LetterString.cpp
#include "LetterString.h"

LetterString::LetterString() : StringBase() {}

LetterString::LetterString(const std::string& str) : StringBase(str) {}

LetterString::LetterString(const LetterString& other) : StringBase(other) {}

void LetterString::shiftLastChar() {
    if (text.length() > 1) {
        char lastChar = text.back();  
        text.pop_back();              
        text = lastChar + text;       
    }
}

std::string LetterString::getShiftedText() const {
    return text;
}

File: LetterString.cs
public class LetterString : StringBase
{
    public LetterString() : base() { }

    public LetterString(string str) : base(str) { }

    public LetterString(LetterString other) : base(other) { }

    public void ShiftLeft()
    {
        if (text.Length > 1)
        {
            char lastChar = text[text.Length - 1]; 
            text = lastChar + text.Substring(0, text.Length - 1); 
        }
    }


    public string GetShiftedText()
    {
        return text;
    }
}
